lib/model/model.go

```diff
@@ -784,6 +787,40 @@ func (m *Model) NeedFolderFiles(folder string, page, perpage int) ([]db.FileInfo
 	return progress, queued, rest, total
 }
 
+// RemoteNeedFolderFiles returns paginated list of currently needed files in
+// progress, queued, and to be queued on next puller iteration, as well as the
+// total number of files currently needed.
+func (m *Model) RemoteNeedFolderFiles(device protocol.DeviceID, folder string, page, perpage int) ([]db.FileInfoTruncated, int, error) {
+	m.pmut.RLock()
+	m.fmut.RLock()
```

Comment:
So this might be correct, but you should double check that the order these are acquired is the same as somewhere else.